drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -73,6 +73,15 @@ def run!(user:)
         processor = container_processor(container)
         raise unknown_field_error(definition) unless processor
 
+        # TODO @MARTHA couple things:
+        # - This doesn't work because the frontend includes the value, even if nil
+        #   - We can't just check for nil values, because that would miss users trying to turn an existing value back to nil
+        #   - We probably don't want to take the time to go into every form processor and do this individually, but there is no other way to know whether the attribute has already been filled out or not
+        #   - Alternatively can we get the frontend to stop sending these values, or is that too risky a change?
+        # - with mapped_form_items I'm trying to be generic to other future use cases where we need to know something about the item's attributes when processing the form. Should we somehow try to combine mapped_form_items with the other maps or is that too risky a change?
+        item = mapped_form_items["#{container}:#{field}"]
```

Comment:
@gigxz I would like to check in about this because I think this is a non-straightforward aspect of the Approach 2 which we settled on. Since we liked Approach 2 for simplicity and ease, before proceeding I want to make sure this doesn't tip us towards a different approach.

Problem:
- In `hud_values_by_container` passed from the frontend, all fields are included, even when the value is nil. So the "raise" condition I've added here incorrectly fails when a user submits a form with non-editable-by-them items on it, even if they haven't actually modified those items
- I think we can't just check for non-nil values, because that would miss turning an existing value back to nil
- In order to compare against the existing saved value, I think we would need to modify every form processor - there is no other way to know whether the attribute has already been saved or not, right?

Possible solutions:
- Approach 2a: modify the behavior of each individual form processor to check if a value has already been saved and if so, if it's different from the value coming from the form submission. This seems more time consuming than is worthwhile for something that is intended to be a temporary solution.
- Approach 2b: modify the behavior of [transformSubmitValues](https://github.com/greenriver/hmis-frontend/blob/release-140/src/modules/form/util/formUtil.ts#L1139) on the frontend so that it doesn't even include keys for items on which the user doesn't have edit permission. (`transformSubmitValues` would need to accept a new param for the current user's ID.)

Reply:
Approach 2c: Backend can just ignore the value if the user doesn't have access.

Take care to save the original raw state in the hud values to the database though.

Reply:
Discussed: lets just have the backend processor ignore the value if it's a field that the user cannot edit